src/app/core/feedback/feedback.service.js

```diff
@@ -56,16 +70,51 @@ const search = async (reqUser, queryParams, query) => {
 	const offset = page * limit;
 
 	// Query for feedback
-	const feedback = await Feedback.textSearch(query, null, limit, offset, sortArr, true, {
+	const feedback = await Feedback.textSearch(
+		query,
+		null,
+		limit,
+		offset,
+		sortArr,
+		true,
+		{
+			path: 'creator',
+			select: ['username', 'organization', 'name', 'email']
+		}
+	);
+
+	return util.getPagingResults(limit, page, feedback.count, feedback.results);
+};
+
+const getFeedbackById = async (feedbackId) => {
+	return Feedback.findOne({
+		_id: mongoose.Types.ObjectId(feedbackId)
+	}).populate({
 		path: 'creator',
-		select: ['name', 'email']
+		select: ['username', 'organization', 'name', 'email']
 	});
+};
 
-	return util.getPagingResults(limit, page, feedback.count, feedback.results);
+const updateFeedbackAssignee = async (feedbackId, assignee) => {
+	const feedback = await getFeedbackById(feedbackId);
+	feedback.assignee = assignee;
+	feedback.updated = Date.now();
+	await feedback.save();
+	return feedback;
+};
+
+const updateFeedbackStatus = async (feedbackId, status) => {
+	const feedback = await getFeedbackById(feedbackId);
+	feedback.status = status;
+	feedback.updated = Date.now();
+	await feedback.save();
```

Comment:
@sshah-asymmetrik is the intent here to return the updated feedback? If so, should we be returning the response from `await feedback.save();` in case there are other mongoose middleware methods that modify the object?

Reply:
Are you concerned about pre/post save hooks?  I'm pretty sure that they are all guaranteed to be completed before the save promise will complete.

Reply:
@jrassa no, I was concerned about whether the response from the `save()` method would be the same as the initial `feedback` object that is being saved ( and currently returned ).

Reply:
@brianghig I misread the code.  I think they would be the same, but agree that we should return the response.

Reply:
@brianghig They _should_ be the same, but I'll update it to use the response from `.save()` just to be extra clear.

Reply:
- [ ] return .save()